cpp/dotenv.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Very simple dotenv implementation in C++,
+ * specifically for the use case of running tests in this repository.
+ *
+ * It tries to read `.env` file in the current directory or a custom file.
+ * If no file is found, no error is thrown, instead the environment variable
+ * is expected to be set in the environment.
+ *
+ * If both the file and the environment variable are set, the environment variable
+ * takes precedence.
+ */
+
+#ifndef DOTENV_H
+#define DOTENV_H
+
+#include <algorithm>
+#include <exception>
+#include <filesystem>
+#include <fstream>
+#include <map>
+#include <stdexcept>
+#include <string>
+
+namespace dotenv {
+
+class Dotenv {
+  private:
+    std::string envFile;
+    std::map<std::string, std::string> envFromFile;
+
+    std::map<std::string, std::string> readEnvFromFile() {
+        std::map<std::string, std::string> env;
+        std::ifstream file(envFile);
+        char charsToRemove[] = {'"', '\''};
+        
+        if (!file) {
+            return env;
+        }
+        
+        for (std::string line; std::getline(file, line);) {
+            const auto pos = line.find('=');
+            if (pos != std::string::npos) {
+                const auto key = line.substr(0, pos);
+                auto value = line.substr(pos + 1);
```

Comment:
Might throw an out_of_range exception if the line ends with =. Do we care?

Reply:
tbh, I don't care. The parser is only for the tests and not meant to be a fully correct env. parser. It works for what we need, so I would keep it simple.